src/metabase/driver/mongo/query_processor.clj

```diff
@@ -158,7 +161,26 @@
 
   DateTimeField
   (->rvalue [this]
-    (str \$ (->lvalue this)))
+    ;; omf: date/time fields using native Mongo operators
+    (let [unit (:unit this)
+      field (:field this)]
+      (case (or unit :default)
+        :default         (->rvalue field)
+        :minute          {:$dateToString {:format "%Y-%m-%d %H:%M" :date (->rvalue field)}}
+        :minute-of-hour  {:$minute (->rvalue field)}
+        :hour            {:$dateToString {:format "%Y-%m-%d %H" :date (->rvalue field)}}
+        :hour-of-day     {:$hour (->rvalue field)}
+        :day             {:$dateToString {:format "%Y-%m-%d" :date (->rvalue field)}}
+        :day-of-week     {:$dayOfWeek (->rvalue field)}
+        :day-of-month    {:$dayOfMonth (->rvalue field)}
+        :day-of-year     {:$dayOfYear (->rvalue field)}
+        :week            {:$dateToString {:format "%Y-%m-%d" :date { :$subtract [ (->rvalue field) {:$multiply [ {:$subtract [{:$dayOfWeek (->rvalue field)}, 1]}, 86400000]}] }}}
+        :week-of-year    {:$add [{:$week (->rvalue field)}, 1]}
+        :month           {:$dateToString {:format "%Y-%m" :date (->rvalue field)}}
+        :month-of-year   {:$month (->rvalue field)}
+        :quarter         {:$dateToString {:format "%Y-%m" :date (->rvalue field)}} 
+        :quarter-of-year {:$substr [{:$add [{:$divide [{:$subtract [{:$month (->rvalue field)} ,1]} ,3]} ,1]} ,0 ,1]}
```

Comment:
keep in mind comments in Lisp languages are whitespace so this is not normally a place where they would go. And they would never be written like `element ,another-element` (space before comment)